### ISSUE TITLE
Error when recreating groups

### ISSUE THREAD

**Issue by CtrlAltDefeat94** [ShokoAnime/ShokoServer#750]
On commit: 691f6b5ecae875dd465134c09d55b8a092d997be

Seems recreating groups broke my groups entirely:
![image](https://user-images.githubusercontent.com/5640470/43019194-6a4022ec-8c5c-11e8-8303-01f8a5fbccc7.png)

Log:
https://gist.github.com/CtrlAltDefeat94/8a3d07f8c1f4a1f6d6dc585b748e0c3a

**Comment by da3dsoul** [comment]
That is one that I can fix

**Comment by da3dsoul** [comment]
I pushed something that might help. If it fails, then there's something up in your DB.

In that case run this on your DB.

```
DELETE FROM GroupFilter WHERE FilterType = 16
```

**Comment by CtrlAltDefeat94** [comment]
Filters are populating again with grouping turned off.
With grouping turned on, I managed to get the error from the first post again, however this time just one, rather than 15 inner exceptions(though I doubt that's progress there heh)

**Comment by da3dsoul** [comment]
Should be fixed now

**Comment by CtrlAltDefeat94** [comment]
When trying to recreate groups with grouping enabled
```[2018-07-22 08:33:22:413] Error|ShokoServiceImplementation.RecreateAllGroups => AnimeGroupCreator.RecreateAllGroups => AnimeGroupCreator.RecreateAllGroups An error occurred while re-creating all groups
[2018-07-22 08:33:22:775] Error|MonoMethod.Invoke => MonoMethod.InternalInvoke => ShokoServiceImplementation.RecreateAllGroups System.NullReferenceException: Object reference not set to an instance of an object
  at System.Linq.Enumerable+<SelectManyIterator>d__167`3[TSource,TCollection,TResult].MoveNext () [0x00060] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at System.Linq.Lookup`2[TKey,TElement].Create[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] keySelector, System.Func`2[T,TResult] elementSelector, System.Collections.Generic.IEqualityComparer`1[T] comparer) [0x00031] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at System.Linq.Enumerable.ToLookup[TSource,TKey,TElement] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] keySelector, System.Func`2[T,TResult] elementSelector, System.Collections.Generic.IEqualityComparer`1[T] comparer) [0x0002a] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at System.Linq.Enumerable.ToLookup[TSource,TKey,TElement] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] keySelector, System.Func`2[T,TResult] elementSelector) [0x00000] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at Shoko.Server.Repositories.Cached.GroupFilterRepository+<>c__DisplayClass29_0.<CalculateAnimeSeriesPerTagGroupFilter>b__8 (System.Int32 key) [0x00000] in /usr/src/app/source/Shoko.Server/Repositories/Cached/GroupFilterRepository.cs:709 
  at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] keySelector, System.Func`2[T,TResult] elementSelector, System.Collections.Generic.IEqualityComparer`1[T] comparer) [0x00096] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] keySelector, System.Func`2[T,TResult] elementSelector) [0x00000] in <521f3fe476f34ea88748e97b3b984d05>:0 
  at Shoko.Server.Repositories.Cached.GroupFilterRepository.CalculateAnimeSeriesPerTagGroupFilter (Shoko.Server.Repositories.NHibernate.ISessionWrapper session) [0x000fb] in /usr/src/app/source/Shoko.Server/Repositories/Cached/GroupFilterRepository.cs:709 
  at Shoko.Server.Tasks.AnimeGroupCreator.UpdateGroupFilters (Shoko.Server.Repositories.NHibernate.ISessionWrapper session) [0x00024] in /usr/src/app/source/Shoko.Server/Tasks/AnimeGroupCreator.cs:190 
  at Shoko.Server.Tasks.AnimeGroupCreator.RecreateAllGroups (Shoko.Server.Repositories.NHibernate.ISessionWrapper session) [0x001ef] in /usr/src/app/source/Shoko.Server/Tasks/AnimeGroupCreator.cs:510 
  at Shoko.Server.Tasks.AnimeGroupCreator.RecreateAllGroups () [0x0000b] in /usr/src/app/source/Shoko.Server/Tasks/AnimeGroupCreator.cs:525 
  at Shoko.Server.ShokoServiceImplementation.RecreateAllGroups (System.Boolean resume) [0x00000] in /usr/src/app/source/Shoko.Server/API/v1/Implementations/ShokoServiceImplementation/ShokoServiceImplementation_Entities.cs:1798

**Comment by CtrlAltDefeat94** [comment]
I just noticed that after recreating groups with grouping enabled(after I did above mentioned), all shows are grouped according to the settings used before the crash

**Comment by da3dsoul** [comment]
WTF. @CtrlAltDefeat I think there's something seriously up with your DB. It's throwing nullpointers in places that should never be null. I'll try to make it run again, but you may end up with strange errors later that seem to make no sense.

**Comment by da3dsoul** [comment]
Try with latest commit

**Comment by bigretromike** [comment]
@da3dsoul the "corrupt db" is in most cases because some errors in server ;-)

**Comment by da3dsoul** [comment]
@ctrlaltdefeat what was the status with latest? I know that you said it doesn't work, but I'd like a stack

**Comment by CtrlAltDefeat94** [comment]
This was from my latest try
https://gist.github.com/CtrlAltDefeat94/ce7226c3421af531ed69cf2a3639f8ce

**Comment by bigretromike** [comment]
@da3dsoul no, no mysql timeout normaly. But I dont recreate groups.

**Comment by CtrlAltDefeat** [comment]
Just a heads up, yall keep @ ing me instead of @CtrlAltDefeat94

**Comment by da3dsoul** [comment]
Wait! Who are you?

**Comment by CtrlAltDefeat** [comment]
Ha, yeah.. Alternatively just pay attention to the actual username of whoever you're pinging :p Good luck on your project!

**Comment by CtrlAltDefeat94** [comment]
Issue appears to be resolved on f548b39444f44ab897fbcf9647fb771d4abdab8c
